app/charts/chart-config-ui-options.ts

```diff
@@ -1,76 +1,78 @@
-import { ChartType, SortingOrder } from "../configurator";
-import { imputationTypes, SortingType } from "../configurator/config-types";
+import {
+  ChartType,
+  ComponentType,
+  SortingOrder,
+  SortingType,
+} from "@/configurator/config-types";
 
 /**
  * This module controls chart controls displayed in the UI.
  * Related to config-types.ts.
  */
 
-// This should match graphQL Schema
-export type DimensionType =
-  | "NumericalMeasure"
-  | "OrdinalMeasure"
-  | "GeoCoordinatesDimension"
-  | "GeoShapesDimension"
-  | "NominalDimension"
-  | "OrdinalDimension"
-  | "TemporalDimension";
+export type EncodingFieldType =
+  // Generic.
+  | "x"
+  | "y"
+  | "segment"
+  // Map.
+  | "baseLayer"
+  | "areaLayer"
+  | "symbolLayer";
```

Comment:
I think it would be good to have

```
type MapEncodings = 'baseLayer' | 'areaLayer' | 'symbolLayer'
type XYEncodings = 'x' | 'y' | 'segment'
type Encodings = MapEncodings | XYEncodings
```

This way we can use the granularity when we need it, and we do not need the comments.